src/k4a_ros_device.cpp

```diff
@@ -58,30 +58,29 @@ K4AROSDevice::K4AROSDevice()
-  this->declare_parameter("depth_mode");
-  this->declare_parameter("color_enabled");
-  this->declare_parameter("color_format");
-  this->declare_parameter("color_resolution");
-  this->declare_parameter("fps");
-  this->declare_parameter("point_cloud");
-  this->declare_parameter("rgb_point_cloud");
-  this->declare_parameter("point_cloud_in_depth_frame");
-  this->declare_parameter("required");
-  this->declare_parameter("sensor_sn");
-  this->declare_parameter("recording_file");
-  this->declare_parameter("recording_loop_enabled");
-  this->declare_parameter("body_tracking_enabled");
-  this->declare_parameter("body_tracking_smoothing_factor");
-  this->declare_parameter("rescale_ir_to_mono8");
-  this->declare_parameter("ir_mono8_scaling_factor");
-  this->declare_parameter("imu_rate_target");
-  this->declare_parameter("wired_sync_mode");
-  this->declare_parameter("subordinate_delay_off_master_usec");
-  this->declare_parameter({depth_raw_topic + compressed_format});
-  this->declare_parameter({depth_raw_topic + compressed_png_level});
-  this->declare_parameter({depth_rect_topic + compressed_format});
-  this->declare_parameter({depth_rect_topic + compressed_png_level});
+  this->declare_parameter("depth_enabled", rclcpp::ParameterValue(true));
+  this->declare_parameter("depth_mode", rclcpp::ParameterValue("NFOV_UNBINNED"));
+  this->declare_parameter("color_enabled", rclcpp::ParameterValue(false));
+  this->declare_parameter("color_format", rclcpp::ParameterValue("bgra"));
+  this->declare_parameter("color_resolution", rclcpp::ParameterValue("720P"));
+  this->declare_parameter("fps", rclcpp::ParameterValue(5));
+  this->declare_parameter("point_cloud", rclcpp::ParameterValue(true));
+  this->declare_parameter("rgb_point_cloud", rclcpp::ParameterValue(false));
+  this->declare_parameter("point_cloud_in_depth_frame", rclcpp::ParameterValue(true));
+  this->declare_parameter("sensor_sn", rclcpp::ParameterValue(""));
+  this->declare_parameter("recording_file", rclcpp::ParameterValue(""));
+  this->declare_parameter("recording_loop_enabled", rclcpp::ParameterValue(false));
+  this->declare_parameter("body_tracking_enabled", rclcpp::ParameterValue(false));
+  this->declare_parameter("body_tracking_smoothing_factor", rclcpp::ParameterValue(0.0f));
+  this->declare_parameter("rescale_ir_to_mono8", rclcpp::ParameterValue(false));
+  this->declare_parameter("ir_mono8_scaling_factor", rclcpp::ParameterValue(1.0f));
+  this->declare_parameter("imu_rate_target", rclcpp::ParameterValue(0));
+  this->declare_parameter("wired_sync_mode", rclcpp::ParameterValue(0));
+  this->declare_parameter("subordinate_delay_off_master_usec", rclcpp::ParameterValue(0));
+  // this->declare_parameter({depth_raw_topic + compressed_format});
+  // this->declare_parameter({depth_raw_topic + compressed_png_level});
+  // this->declare_parameter({depth_rect_topic + compressed_format});
+  // this->declare_parameter({depth_rect_topic + compressed_png_level});
```

Comment:
I'm not really sure here, these were parameters with their name being the topic name itself? What is the use here? Where are these used?

Reply:
Honestly, not sure either. I think this was done as an alternative to topic remapping, but likely should be removed.